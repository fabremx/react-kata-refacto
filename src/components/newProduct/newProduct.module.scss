@import "../../assets/styles/colors.scss";

$green: #488f4f;

.photo div {
  font-size: 11px;
  font-weight: 600;
  background-color: $green;
  color: white;
  text-align: center;
  padding: 5px;
  border-radius: 3px;
}

.name span {
  color: $green;
  font-weight: 600;
}
