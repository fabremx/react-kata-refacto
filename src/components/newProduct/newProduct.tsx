import { ReactElement } from "react";
import baseProductStyles from "../baseProduct/baseProduct.module.scss";
import styles from "./newProduct.module.scss";

interface Props {
  title: string;
}

export function NewProduct({ title }: Props): ReactElement {
  return (
    <>
      <td
        className={`${baseProductStyles.photo} ${styles.photo}`}
        data-testid="photoNewProduct"
      >
        <div>New product</div>
      </td>

      <td
        className={`${baseProductStyles.name} ${styles.name}`}
        data-testid="titleNewProduct"
      >
        {title} <span>(New product)</span>
      </td>
    </>
  );
}
