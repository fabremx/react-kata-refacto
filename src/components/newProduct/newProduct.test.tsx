import { render } from "@testing-library/react";
import { NewProduct } from "./newProduct";

describe("NewProduct", () => {
  it("should render correctly the component", () => {
    // Given
    const tableRow = document.createElement("tr");
    // When
    const { asFragment } = render(<NewProduct title={"Title"} />, {
      container: document.body.appendChild(tableRow),
    });
    // Then
    expect(asFragment()).toMatchSnapshot();
  });
});
