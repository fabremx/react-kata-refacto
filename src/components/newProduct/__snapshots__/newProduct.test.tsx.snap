// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProduct should render correctly the component 1`] = `
<DocumentFragment>
  <div>
    New product
  </div>
  Title 
  <span>
    (New product)
  </span>
</DocumentFragment>
`;
